internal/fuzzbinder/fuzzscraper.go

```diff
@@ -0,0 +1,265 @@
+}
+
+// getBucket gets a GCS bucket given its name.
+func getBucket(bucketName string) (*storage.BucketHandle, error) {
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("storage.NewClient: %v", err)
+	}
+	defer client.Close()
+
+	bucket := client.Bucket(bucketName)
+	return bucket, nil
+}
+
+// listBlobs gets all the objects in a GCS bucket under a given relative path.
+func listBlobs(bucket *storage.BucketHandle, relativePath string) ([]string, error) {
+	var blobs []string
+	ctx := context.Background()
+	query := &storage.Query{Prefix: relativePath}
+	objects := bucket.Objects(ctx, query)
+	for {
+		attrs, err := objects.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return nil, fmt.Errorf("Bucket.Objects: %v", err)
+		}
+		blobs = append(blobs, attrs.Name)
+	}
+	return blobs, nil
+}
+
+// getBlob gets the file reader of a blob in a GCS bucket.
+func getBlob(bucket *storage.BucketHandle, blobName string) (*storage.Reader, error) {
+	ctx := context.Background()
+	reader, err := bucket.Object(blobName).NewReader(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("Object(%q).NewReader: %v", blobName, err)
+	}
+	defer reader.Close()
+	return reader, nil
+}
+
+// getRev gets the revision of a source code given a source map file.
+func getRev(reader *storage.Reader, project string) (string, error) {
+	var payload map[string](map[string]string)
+	content, _ := io.ReadAll(reader)
+	err := json.Unmarshal(content, &payload)
+	if err != nil {
+		return "", fmt.Errorf("unmarshal: %v", err)
+	}
+	rev := payload[fmt.Sprintf("/src/%s", project)]["rev"]
+	return rev, nil
+}
+
+// parseCoverage gets the branch coverage and line coverage from a coverage report summary.
+func parseCoverage(reader *storage.Reader) (map[string]float64, map[string]float64, error) {
+	var payload CoverageSummary
+	content, _ := io.ReadAll(reader)
+	err := json.Unmarshal(content, &payload)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unmarshal: %v", err)
+	}
+	return payload.Data[0].Totals["branches"], payload.Data[0].Totals["lines"], nil
+}
+
+// getLogs gets the log files of a fuzz-target on a given day.
+func getLogs(date string, project string, fuzzTarget string, fuzzEngine string, sanitizer string) (*storage.BucketHandle, []string) {
+	logsBucket := fmt.Sprintf("%s-logs.clusterfuzz-external.appspot.com", project)
+	bucket, err := getBucket(logsBucket)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fuzzengine := strings.ToLower(fuzzEngine)
+	relativePath := fmt.Sprintf("%s_%s_%s/%s_%s_%s/%s", fuzzEngine, project, fuzzTarget, fuzzengine, sanitizer, project, date)
+	blobs, err := listBlobs(bucket, relativePath)
+	if err != nil {
+		log.Fatal(err)
+	}
+	return bucket, blobs
+}
+
+// GetFuzzedHash gets the revision (a hash) of the source code for which a coverage report
+// was generated on a given day.
+func GetFuzzedHash(date string, project string) string {
```

Comment:
Idea: you could introduce a data type for the hash of a revision, something like `RevisionHash`.

Reply:
I changed the naming of the rev to `revisionHash` in all the code since it's more clear. However, I don't understand why we need a new data type  `RevisionHash` since it's a string (unless there is some verification we want to do).